src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,608 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""
+==================================================================================
+  Compute Interference caused by a grant for all the incumbent types
+  APIs in this file are used by IAP and Aggregate Interference Reference Models
+
+  The main routines are:
+
+    computeInterferencePpaGwpzPoint 
+    computeInterferenceEsc 
+    computeInterferenceFss 
+    computeInterferenceFssBlocking 
+    calculateInterference 
+
+  The common utility APIs are:
+
+    getAllGrantInformationFromCbsdDataDump 
+    findOverlappingGrantsInsideNeighborhood
+    getProtectedChannels
+
+  The routines return a interference caused by a grant in the neighborhood of 
+  FSS/GWPZ/PPA/ESC incumbent types
+==================================================================================
+"""
+import numpy as np
+from reference_models.antenna import antenna
+from reference_models.geo import vincenty
+from reference_models.propagation import wf_itm
+from reference_models.propagation import wf_hybrid
+from collections import namedtuple
+from enum import Enum
+from shapely.geometry import Point as SPoint
+from shapely.geometry import Polygon as SPolygon
```

Comment:
But by the way why do you need these routines here.
Please remove all import that you do not use.

Reply:
Removed